specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+    [RequestRestartAsync](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.coreapplication.requestrestartasync?view=winrt-18362)
+    method, which allows an app to terminate and restart itself, and to
+    provide an arbitrary command-line string for the restarted instance.
+
+2.  **Restart-me-after-termination**. The Win32 API
+    [RegisterApplicationRestart](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-registerapplicationrestart)
+    enables an app to register itself to be restarted after termination,
+    and to provide an arbitrary command-line string for the restarted
+    instance. The reasons for termination include app crash or hang, app
+    update, or system update. There's also a matching
+    [UnregisterApplicationRestart](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-unregisterapplicationrestart)
+    API.
+
+There are 2 related Win32 APIs that focus on recovery prior to restart:
+
+1.  **Let-me-do-something-before-termination**. An app can call
+    [RegisterApplicationRecoveryCallback](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-registerapplicationrecoverycallback)
+    to register a callback for the system to call before terminating the
+    app. If an application encounters an unhandled exception or becomes
+    unresponsive, Windows Error Reporting (WER) calls the specified
+    recovery callback, where the app can save state information. The
+    system pings the app every n seconds to make sure that it hasn't
+    hung in its callback. The app can specify the ping interval when it
+    registers the callback. In its recovery callback, the app can call
+    RegisterApplicationRestart a second time, to update the
+    command-line.
+
+2.  **Recovery-in-progress**. While the app is doing work in its
+    recovery callback, the app must periodically call
+    [ApplicationRecoveryInProgress](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-applicationrecoveryinprogress)
+    \-- if it doesn't call within the registered ping interval, WER will
+    terminate the process.
+
+| App Type                                                            | Applicable Restart Mechanism                                           | Gap                                                                                                                  |
+| ------------------------------------------------------------------- | ---------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------- |
+| Win32 (unpackaged), Win32/Centennial (packaged), WinUI (Centennial) | RegisterApplicationRestart for OS + app update/restart/crash scenarios | Apps can register with the OS to restart in specific app/OS states, but cannot restart on-command in a healthy state |
+
+## Existing API Details
+
+<table>
```

Comment:
Document style comment: API specs shouldn't use HTML. We want the API spec to roll into the docs, and the docs are Markdown-only. Markdown isn't set up well for tall table rows like this, but that's actually a ploy to discourage tables with tall rows. This API in the docs will become a page, so better to just make it a section.

Reply:
Yes, the reason for the HTML was that Markdown doesn't support tables with bullets like this. However, given this suggestion I can refactor it into its own section.

Reply:
RECOMMEND: We'll move it to a section, but generally we need to revamp the template to AVOID tables.

Reply:
Moved to sections, removed tables